src/transform/plugins/term/termDefinitions.ts

```diff
@@ -121,31 +130,53 @@ function processTermDefinition(
         state.env.terms[':' + label] = title;
     }
 
-    const termNodes = [];
-
+    const map: [number, number] = [startLine, endLine];
     token = new state.Token('template_open', 'template', 1);
+    token.map = map;
     token.attrSet('id', ':' + label + '_template');
-    termNodes.push(token);
+    token.attrSet('label', label);
 
-    token = new state.Token('term_open', 'dfn', 1);
+    state.tokens.push(token);
+
+    token = new state.Token('dfn_open', 'dfn', 1);
     token.attrSet('class', 'yfm yfm-term_dfn');
     token.attrSet('id', ':' + label + '_element');
     token.attrSet('role', 'tooltip');
-    termNodes.push(token);
 
-    termNodes.push(...md.parse(title, {}));
+    state.tokens.push(token);
 
-    token = new state.Token('term_close', 'dfn', -1);
-    termNodes.push(token);
+    const withOffset = md.parse(title, {}).map((el) => {
```

Comment:
let's give it more meaningfull name, also let's not increase nesting depth and move it out to the root scope.